app_dart/lib/src/model/github/checks.dart

```diff
@@ -52,3 +52,96 @@ class CheckRun {
 
   Map<String, dynamic> toJson() => _$CheckRunToJson(this);
 }
+
+/// Data model for a `merge_group` event.
+///
+/// The event may request one of two actions:
+///
+/// * `checks_requested`: request to create checks and kick of CI tasks.
+/// * `destroyed`: a notification that a `merge_group` is no longer neede and CI
+///   jobs can be stopped.
+///
+/// See more:
+///  * https://docs.github.com/en/webhooks/webhook-events-and-payloads#merge_group.
+@JsonSerializable(fieldRename: FieldRename.snake)
+class MergeGroupEvent extends HookEvent {
+  MergeGroupEvent({
+    required this.action,
+    required this.mergeGroup,
+    this.repository,
+    this.sender,
+  });
+
+  factory MergeGroupEvent.fromJson(Map<String, dynamic> input) => _$MergeGroupEventFromJson(input);
+
+  String action;
+  MergeGroup mergeGroup;
+  Repository? repository;
+  User? sender;
+
+  Map<String, dynamic> toJson() => _$MergeGroupEventToJson(this);
+}
+
+/// Data model for a `merge_group`.
+///
+/// This object is supplied by [MergeGroupEvent].
+@JsonSerializable(fieldRename: FieldRename.snake)
+class MergeGroup {
+  const MergeGroup({
+    required this.headSha,
+    required this.headRef,
+    required this.baseSha,
+    required this.baseRef,
+    required this.headCommit,
+  });
+
+  factory MergeGroup.fromJson(Map<String, dynamic> input) => _$MergeGroupFromJson(input);
+
+  /// The commit SHA generated for the candidate PR to be merged.
+  ///
+  /// This is the SHA that must be checked by the CI.
+  final String headSha;
+
+  /// The ref to the read-only branch generated by Github for the respective PR.
+  final String headRef;
+
+  /// The SHA of the parent commit of the [headSha] commit.
```

Comment:
Maybe:
```
///
/// This is the SHA head of a merge queue protect branch
```

Reply:
Used github docs verbatim for now. We can clarify when we know more about what this is.